c2rust-analyze/src/rewrite/ty.rs

```diff
@@ -266,20 +348,37 @@ fn mk_rewritten_ty<'tcx>(
 struct HirTyVisitor<'a, 'tcx> {
     asn: &'a Assignment<'a>,
     acx: &'a AnalysisCtxt<'a, 'tcx>,
-    rw_lcx: LabeledTyCtxt<'tcx, RewriteLabel>,
+    rw_lcx: LabeledTyCtxt<'tcx, RewriteLabel<'tcx>>,
     mir: &'a Body<'tcx>,
     hir_rewrites: Vec<(Span, Rewrite)>,
     hir_span_to_mir_local: HashMap<Span, rustc_middle::mir::Local>,
 }
 
+fn adt_ty_rw<S>(
+    adt_def: &AdtDef,
+    lifetime_params: &[OriginArg],
+    substs: &&List<GenericArg>,
```

Comment:
```suggestion
    substs: &List<GenericArg>,
```
Haven't tested this, but I wouldn't expect it to break anything, since there's an implicit coercion from `&&T` to `&T`